python/ray/serve/tests/test_cli.py

```diff
@@ -272,6 +272,26 @@ def check_for_failed_deployment():
     wait_for_condition(check_for_failed_deployment, timeout=2)
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
+def test_status_invalid_runtime_env(ray_start_stop):
+    """Deploys a config file with invalid runtime env and checks status.
+
+    get_status() should not throw error (meaning REST API returned 200 status code) and
+    the status be deploy failed."""
+
+    config_file_name = os.path.join(
+        os.path.dirname(__file__), "test_config_files", "bad_runtime_env.yaml"
+    )
+
+    subprocess.check_output(["serve", "deploy", config_file_name])
+
+    def check_for_failed_deployment():
+        app_status = ServeSubmissionClient("http://localhost:52365").get_status()
+        return app_status["app_status"]["status"] == "DEPLOY_FAILED"
+
+    wait_for_condition(check_for_failed_deployment)
```

Comment:
I'd suggest also checking somewhere in the test that the user-friendly runtime env error output (e.g. s3 bucket not found, or pip package does not exist) appears in the expected place.  (You can just check for existence of a relevant substring)